File: Programmers/짝지어 제거하기.py
def solution(s):
    answer = 0
    stack = []

    for i in range(len(s)):
        try:
            if i > 0:
                if stack[-1] != s[i]:
                    stack.append(s[i])
                else:
                    stack.pop()
            else:
                stack.append(s[i])
        except IndexError:
            stack.append(s[i])

    if len(stack) == 0:
        answer = 1

    return answer

File: Programmers/짝지어 제거하기.cpp
#include <bits/stdc++.h>

using namespace std;

stack<char> st;
int solution(string s)
{
    int answer = 0;
    for(int i=0; i<s.size(); i++){
        if(st.size() > 0 && st.top() == s[i]) st.pop();
        else st.push(s[i]);
    }
    answer = !st.size();

    return answer;
}
